draft-ietf-quic-qpack.md

```diff
@@ -685,10 +713,37 @@ Header data is prefixed with two integers, `Largest Reference` and `Base Index`.
 
 `Largest Reference` identifies the largest absolute dynamic index referenced in
 the block.  Blocking decoders use the Largest Reference to determine when it is
-safe to process the rest of the block.
+safe to process the rest of the block.  If Largest Reference is greater than
+zero, the encoder transforms it as follows before encoding:
+
+~~~
+   LargestReference = LargestReference mod 2*MaxEntries + 1
+~~~
+
+The decoder reconstructs the Largest Reference using the following algorithm:
+
+~~~
+   if LargestReference > 0:
+      LargestReference -= 1
+      CurrentWrapped = TableLargestAbsoluteIndex mod 2*MaxEntries
+
+      if CurrentWrapped >= LargestReference + MaxEntries:
```

Comment:
It seems this range is one too long: `[ LargestReference - MaxEntries, LargestReference + MaxEntries ]` is `2 * MaxEntries + 1` long, while the actual valid range is at most `2 * MaxEntries`.  I don't yet know whether this is a problem, as I have to get used to this version of the logic.

Reply:
OK.  As long as it works.

Reply:
I think the phantom entry is `decoder base - MaxEntries`.  The encoder can be a full MaxEntries *ahead* of the decoder, but cannot reference a header MaxEntries less than decoder base, it has necessarily been evicted.  

As such, the second conditional should read 

`else if CurrentWrapped + MaxEntries < LargestReference`

My verification script had a bug that prevented it from testing this exact edge case!  It is the case where CurrentWrapped == MaxEntries - 1 and LargestReference = 2*MaxEntries - 1.  CurrentWrapped + MaxEntries identifies an entry the decoder does not yet have.